public/app/features/alerting/unified/hooks/useAbilities.ts

```diff
@@ -49,53 +58,137 @@ export enum AlertmanagerAction {
-    [AlertSourceAction.ViewAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleRead)],
-    [AlertSourceAction.UpdateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleUpdate)],
-    [AlertSourceAction.DeleteAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleDelete)],
-    // -- External alert rules (Mimir / Loki / etc) --
-    // for these we only have "read" and "write" permissions
-    [AlertSourceAction.CreateExternalAlertRule]: [
+/**
+ * This one will check for alerting abilities that don't apply to any particular alert source or alert rule
+ */
+export const useAlertingAbilities = (): Abilities<AlertingAction> => {
+  const canExportGMARules =
+    ctx.hasPermission(AccessControlAction.AlertingProvisioningRead) ||
+    ctx.hasPermission(AccessControlAction.AlertingProvisioningReadSecrets);
+
+  return {
+    // internal (Grafana managed)
+    [AlertingAction.CreateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleCreate)],
+    [AlertingAction.ViewAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleRead)],
+    [AlertingAction.UpdateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleUpdate)],
+    [AlertingAction.DeleteAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleDelete)],
+    [AlertingAction.ExportGrafanaManagedRules]: [AlwaysSupported, canExportGMARules],
+
+    // external
+    [AlertingAction.CreateExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
-    [AlertSourceAction.ViewExternalAlertRule]: [
+    [AlertingAction.ViewExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalRead),
     ],
-    [AlertSourceAction.UpdateExternalAlertRule]: [
+    [AlertingAction.UpdateExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
-    [AlertSourceAction.DeleteExternalAlertRule]: [
+    [AlertingAction.DeleteExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
   };
+};
+
+export const useAlertingAbility = (action: AlertingAction): Ability => {
+  const allAbilities = useAlertingAbilities();
+  return allAbilities[action];
+};
+
+/**
+ * This hook will check if we support the action and have sufficient permissions for it on a single alert rule
+ */
+export function useAlertRuleAbility(rule: CombinedRule, action: AlertRuleAction): Ability {
+  const abilities = useAllAlertRuleAbilities(rule);
+
+  return useMemo(() => {
+    return abilities[action];
+  }, [abilities, action]);
+}
+
+export function useAlertRuleAbilities(rule: CombinedRule, actions: AlertRuleAction[]): Ability[] {
```

Comment:
I don't see any place where we are using this hook. Is like an alternative for the  `useAlertRuleAbility` if you want to use a list of actions at once?

Reply:
That was the original intention but it seemed to add a larger surface for mocking (we should do less of that, but that's another conversation).

I tried comparing one single `useAlertRuleAbilities()` hook to multiple `useAlertRuleAbility()` hook calls but both read equally good. With the `useAlertRuleAbilities()` you have to make sure to desctruct from the Array in the right index or it will cause bugs.

I'm thinking of just not doing any plural hooks, I've left this one in in case we change our minds but it's probably safer to reduce our code surface and choose the hook with the least potential for bugs.